### DISCUSSION TITLE
I need to move to jackson 2.17.0

### DISCUSSION THREAD

**Discussion by rdykiel** [victools/jsonschema-generator#489] [General]
I'd like to use your library in an in-house project but need it to move to jackson 2.17.0
I cloned the repository and made the change. 'mvn package' passed.

I encountered the following error trying 'mvn install':

[INFO] Java JSON Schema Generator ......................... FAILURE [  7.570 s]
...
[ERROR] Failed to execute goal org.moditect:moditect-maven-plugin:1.0.0.RC3:add-module-info (add-module-infos) on project jsonschema-generator: File jsonschema-generator-4.37.0-SNAPSHOT.jar is already modular -> [Help 1]

I'm unable to push the change to create a PR, so I'd appreciate a solution to at least be able to test locally. Thanks

Here is my change:
commit 6dcd1a2d20a7c933b5b9afda9626324037a71f58 (HEAD -> jackson2.17.0)
Author: Richard Dykiel <rdykiel@voltdb.com>
Date:   Tue Oct 29 13:46:58 2024 -0400

    Move to jackson 2.17.0

diff --git a/jsonschema-generator-parent/pom.xml b/jsonschema-generator-parent/pom.xml
index af5d7a9..346cfb6 100644
--- a/jsonschema-generator-parent/pom.xml
+++ b/jsonschema-generator-parent/pom.xml
@@ -151,7 +151,7 @@

         
         <version.classmate>1.5.1</version.classmate>
-        <version.jackson>2.14.2</version.jackson>
+        <version.jackson>2.17.0</version.jackson>
         <version.slf4j>2.0.3</version.slf4j>
         
         <version.jsonassert>1.5.1</version.jsonassert>

**Comment by rdykiel** [comment]
I was able to run 'mvn install' by disabling the moditec step in this file:

`diff --git a/jsonschema-generator/pom.xml b/jsonschema-generator/pom.xml
index b89ea64..8a4b935 100644
--- a/jsonschema-generator/pom.xml
+++ b/jsonschema-generator/pom.xml
@@ -51,10 +51,12 @@
             <plugin>
                 <artifactId>maven-javadoc-plugin</artifactId>
             </plugin>
+            
         </plugins>
     </build>

`

**Reply by CarstenWickner** [reply]
Hi @rdykiel,

Have you tried `mvn clean install`?
The error message mentioning that the artifact "is already modular" sounds like there is some left-over from a previous execution.

--------

That being said, why would you have to build your own version of the library just to use a higher jackson version?
Couldn't you ignore the transitive dependency and include your own?
```xml
<dependency>
    <groupId>com.github.victools</groupId>
    <artifactId>jsonschema-generator</artifactId>
    <version>4.36.0</version>
    <exclusions>
        <exclusion>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>*</artifactId>
        </exclusion>
    </exclusions>
</dependency>
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-databind</artifactId>
    <version>2.17.0</version>
</dependency>
```

**Reply by rdykiel** [reply]
Thanks for the reply: will try 'mvn clean install', and the transitive dependency. Hope the latter works in my project.
I confess I'm no maven expert :-(

**Reply by rdykiel** [reply]
mvn clean install worked. I will remember to use the 'clean' option after I did a 'package'...
Wow the transitive dependency worked thanks again!

We can consider this discussion resolved. I may open a separate discussion on a possible bug I encountered.

**Comment by CarstenWickner** [comment]
As per the response above:
> Have you tried `mvn clean install`?
> The error message mentioning that the artifact "is already modular" sounds like there is some left-over from a previous execution.

--------

> That being said, why would you have to build your own version of the library just to use a higher jackson version?
> Couldn't you ignore the transitive dependency and include your own?
```xml
<dependency>
    <groupId>com.github.victools</groupId>
    <artifactId>jsonschema-generator</artifactId>
    <version>4.36.0</version>
    <exclusions>
        <exclusion>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>*</artifactId>
        </exclusion>
    </exclusions>
</dependency>
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-databind</artifactId>
    <version>2.17.0</version>
</dependency>
```
